packages/app/src/cli/services/app/select-app.ts

```diff
@@ -47,7 +47,17 @@ export function remoteAppConfigurationExtensionContent(
     if (!configSpec) return
     const configString = module.config
     if (!configString) return
-    const config = configString ? JSON.parse(configString) : {}
+
+    let config
+    if (configString) {
+      if (typeof configString === 'string') {
+        config = JSON.parse(configString)
+      } else {
+        config = configString
+      }
+    } else {
+      config = {}
+    }
```

Comment:
You're right about the inconsistent checks! I believe it can still be a string or object, and this is coercing to an object. Maybe this would be better:

```suggestion
    const config = (typeof configString === 'string') ? JSON.parse(configString) : configString
```

But perhaps there's a way of simplifying further, where something that might be an object isn't called a string.